### PR TITLE
Automatically generate LinuxMain.swift using Sourcery

### PR DESCRIPTION
**Pull request by jpsim**
This adds 6 tests that were accidentally not being run on Linux:

* LineLengthConfigurationTests.testLineLengthConfigurationInitialiserSetsIgnoresComments
* LineLengthConfigurationTests.testLineLengthConfigurationInitialiserSetsIgnoresFunctionDeclarations
* LineLengthConfigurationTests.testLineLengthConfigurationThrowsOnBadConfigValues
* LineLengthRuleTests.testLineLengthWithIgnoreCommentsEnabled
* LineLengthRuleTests.testLineLengthWithIgnoreFunctionDeclarationsEnabled
* RegionTests.testSeveralRegionsFromSeveralCommands

### CODE REVIEW

File: .sourcery/LinuxMain.stencil [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//
+//  LinuxMain.swift
+//  SwiftLint
+//
+//  Created by JP Simard on 12/11/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
```
Comment (marcelofabri) [inline .sourcery/LinuxMain.stencil]: should we add a note that this file is automatically generated and not modified directly?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,8 +28,10 @@ $ swift test
 $ make docker_test
 ```
 
-XCTest functions that are added need to be mirrored in the `allTests` static var
-in the test class extensions at the bottom of the test files.
+If XCTest cases or functions are added/removed/renamed, you'll need to run
```
Comment (marcelofabri) [inline CONTRIBUTING.md]: What do you think about running it automatically in a build phase?

File: Makefile [hunk 3]
```diff
@@ -28,6 +28,11 @@ VERSION_STRING=$(shell /usr/libexec/PlistBuddy -c "Print :CFBundleShortVersionSt
 all: bootstrap
 	$(BUILD_TOOL) $(XCODEFLAGS) build
 
+update_linux_tests:
+	sourcery --sources Tests --templates .sourcery --output .sourcery
```
Comment (marcelofabri) [inline Makefile]: should we verify the version to avoid using different versions?

File: .sourcery/LinuxMain.stencil [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// swiftlint:disable line_length file_length
+
+{% for type in types.classes|based:"XCTestCase" %}
+extension {{ type.name }} {
+    static var allTests: [(String, ({{ type.name }}) -> () throws -> Void)] = [
+    {% for method in type.methods where method.parameters.count == 0 and method.shortName|hasPrefix:"test" %}    ("{{ method.shortName }}", {{ method.shortName }}){% if not forloop.last %},{% endif %}
+    {% endfor %}]
+}
+{% endfor %}
```
Comment (marcelofabri) [inline .sourcery/LinuxMain.stencil]: can we add an empty line between `extension`s?

**Comment by jpsim** [comment]
Closing this PR, since #1592 is a superset. @marcelofabri will address all your feedback there.

**Pull request closed without merging**